### PR TITLE
web: don't show trigger button next to Tiltfile, we currently have no way to force update it

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @hyu, @nicks,

Please review the following commits I made in branch maiamcc/cant-trigger-tiltfile:

4671a148c6a3f88e66d295bb5cd9783a5cc03f7c (2019-11-05 17:59:19 -0500)
web: don't show trigger button next to Tiltfile, we currently have no way to force update it

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: web/src/Sidebar.test.tsx [hunk 1]
```diff
@@ -246,4 +246,31 @@ describe("sidebar", () => {
 
     expect(tree).toMatchSnapshot()
   })
+
+  it("disables trigger button for Tiltfile", () => {
```
Comment (nicks) [inline web/src/Sidebar.test.tsx]: can you elaborate on the value of this test? you don't seem to be asserting anything

File: web/src/SidebarTriggerButton.tsx [hunk 2]
```diff
@@ -46,6 +46,19 @@ export default class SidebarTriggerButton extends PureComponent<
 > {
   render() {
     let props = this.props
+    if (props.resourceName == "(Tiltfile)") {
```
Comment (nicks) [inline web/src/SidebarTriggerButton.tsx]: the protobuf has an isTiltfile property, can you use that instead?

Comment (maiamcc) [reply web/src/SidebarTriggerButton.tsx]: ooh that's much better, will try that, ty

File: web/src/Sidebar.test.tsx (hunk 1 above)
Comment (maiamcc) [reply web/src/Sidebar.test.tsx]: I mean I'm matching against a snapshot in which the trigger button is `disabled` and has the appropriate tooltip. Is your question about what I am/am not expecting to be in the snapshot, or about the ethos of snapshot tests more broadly?

Comment (nicks) [reply web/src/Sidebar.test.tsx]: this test is a pretty good model:
https://github.com/windmilleng/tilt/blob/master/web/src/SidebarTriggerButton.test.tsx#L38

**Pull request merged**